src/btree/bt_read.c

```diff
@@ -125,17 +128,258 @@ __instantiate_birthmarks(WT_SESSION_IMPL *session, WT_REF *ref)
+    __wt_btcur_open(&cbt);
+
+    WT_ERR(__wt_scr_alloc(session, 0, &las_prepares));
+    WT_STAT_CONN_INCR(session, cache_read_lookaside_skew_old);
+    WT_STAT_DATA_INCR(session, cache_read_lookaside_skew_old);
+    if (WT_SESSION_IS_CHECKPOINT(session)) {
+        WT_STAT_CONN_INCR(session, cache_read_lookaside_skew_old_checkpoint);
+        WT_STAT_DATA_INCR(session, cache_read_lookaside_skew_old_checkpoint);
+    }
+
+    /* Open a lookaside table cursor. */
+    __wt_las_cursor(session, &cursor, &session_flags);
+
+    /*
+     * The lookaside records are in key and update order, that is, there will be a set of in-order
+     * updates for a key, then another set of in-order updates for a subsequent key. We find the
+     * most recent of the updates for a key and then insert that update into the page, then all the
+     * updates for the next key, and so on. In case if the birthmark record exists for that key,
+     * then insert birthmark record into the page.
+     *
+     * The search starts with minimum LAS key of the page until the maximum LAS key (both included).
+     * As we are going to instantiate only one record for a key and that too is a most recent one.
+     * To find out the most recent record for a key, we use the search_near cursor function with
+     * both maximum timestamp and transaction id as part of the key search. This search can return
+     * either the last record of the search key or the first record of the next key.
+     *
+     * Once we instantiate the most recent record for a key, the search continues to the next key.
+     * since we're pointing at the max timestamp and transaction id pairing for a given key, the
+     * next entry is guaranteed to be for the next key in the LAS and again set the search for the
+     * next key with maximum timestamp and transaction id.
+     */
+    cache->las_reader = true;
+    __wt_readlock(session, &cache->las_sweepwalk_lock);
+    cache->las_reader = false;
+    first_scan = true;
+    locked = true;
+    cursor->set_key(cursor, las_btree_id, &ref->page_las->min_las_key, WT_TS_MAX, WT_TXN_MAX);
+    for (; ret == 0; ret = cursor->next(cursor)) {
+        if (!first_scan) {
+            WT_ERR(
+              cursor->get_key(cursor, &las_btree_id, &next_las_key, &las_timestamp, &las_txnid));
+
+            /* Set the key to check with maximum timestamp and transaction id.*/
+            cursor->set_key(cursor, las_btree_id, &next_las_key, WT_TS_MAX, WT_TXN_MAX);
+        }
+        first_scan = false;
+
+        WT_ERR(cursor->search_near(cursor, &exact));
+        /*
+         * check whether the search_near returned the same key or not? In case if it returns next
+         * key, continue with the previous record.
+         */
+        if (exact > 0)
+            WT_ERR(cursor->prev(cursor));
+
+        WT_ERR(cursor->get_key(cursor, &las_btree_id, &las_key, &las_timestamp, &las_txnid));
+
+        /* Stop before crossing over to the next btree */
```

Comment:
Style: All the one-line comments in here should be complete sentences, including starting with a capital letter and ending in a period please.

Reply:
Fixed.